iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBDataRegionAsyncConnector.java

```diff
@@ -543,12 +549,24 @@ public synchronized void close() {
     isClosed.set(true);
 
     retryConnector.close();
-    clearRetryEventsReferenceCount();
 
     if (tabletBatchBuilder != null) {
       tabletBatchBuilder.close();
     }
 
+    // ensure all on-the-fly requests have been handled
+    while (hasPendingRequests()) {
+      try {
+        Thread.sleep(50);
+      } catch (final InterruptedException e) {
+        LOGGER.warn(
+            "Interrupted when connector {} waiting for handling pending requests, remaining {} pending requests",
```

Comment:
Plz make sure that printing this will output the connector's id

Reply:
Which ID does "connector id" refer to, the Java object ID or the pipe subtask ID?

Reply:
the pipe subtask ID